pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -984,4 +984,38 @@ self: super: {
     };
   }).override { language-c = self.language-c_0_7_0; };
 
+  hocker =
+    overrideCabal
+      # Not checking because it's failing on a test that needs a data
+      # file not included in its source distribution; this will be
+      # removed when that is fixed
+      ( dontCheck super.hocker )
+      ( oldDerivation: {
+          testToolDepends =
+            (oldDerivation.testToolDepends or []) ++[ pkgs.nix ];
```

Comment:
I guess the package should declare its dependency on Nix, i.e. by depending on http://hackage.haskell.org/package/nix-paths?

Reply:
The only utility from the hocker package that depends on Nix (other than the tests) is `docker2nix`; I think that's clear enough for now. I agree that a build-time dependency on Nix is better than a run-time dependency, so thank you for pointing me at `nix-paths`. Here are my reasons why I don't think we should gate this PR on that, though:

- I don't entirely understand how `nix-paths` works; it looks like it has magic constants in it but I don't understand where those come from; a minimal README.md for the project would help a lot I think (note that if you explain it all to me, I'm also happy to submit a PR with a README.md contribution)
- The utility that we need for `docker2nix` is `nix-hash`, which is not provided by `nix-paths`. I can submit a PR to fix that but I don't think we should gate this PR on that and I also want to understand how `nix-paths` works first before contributing more functionality

Reply:
http://hackage.haskell.org/package/nix-paths-1.0.1 adds support for `nix-hash`. The paths are determined are `configure`-time by Cabal. Look at `Setup.lhs` for the implementation.

Reply:
Thanks for adding support for `nix-hash` @peti. I'll probably get to that soon. In the meantime can I get a review of the other parts of this PR?

I don't think using `nix-paths` is a blocker for this PR but if it is, could you say so?

Reply:
I would not call the `hocker`-related overrides a *blocker*. That's too strong a term, IMHO. What I can say though is that *I* won't merge the PR as-is. Other Nixpkgs maintainers may very well feel differently about it.

Reply:
Okay, thanks for clarifying!

I'd like to clarify my understanding of `nix-paths` (would accept a PR adding a README.md with anything I learn? That may help newcomers and provide a reference to point people at).

I see now that `Setup.lhs` discovers the absolute path to, say, `nix-hash` using `lookupProgram` which is great because if `nix-hash` is not available, it becomes a compile-time error and not a run-time error. Doing this removes the need for the `wrapProgram` invocations but it does require that `nix-paths` is included in the `buildInputs` of `hocker`'s derivation (as a build-time _and_ run-time dependency). Do I have this right?

Additionally, I think the generated derivation for the `nix-paths` hackage package only includes the Nix package in the `nativeBuildInputs`; shouldn't it be in the `propagatedBuildInputs` so that the same Nix used as a build-time dependency is included in the run-time closure of any program that depends on `nix-paths`?

Also, the generated derivation for the `nix-paths` hackage package restricts the hydra build to Linux: `hydraPlatforms = [ "i686-linux" "x86_64-linux" ];` what's the reason for this? `hocker` is usable on Darwin systems, will the `hydraPlatforms` line prevent `hocker` from being integrated on a Darwin worker?